ethereum_history_api/circuits/lib/src/misc/arrays.nr

```diff
@@ -31,36 +31,49 @@ pub(crate) fn memcpy_up_to_length<N, M>(dest: &mut [u8; N], src: [u8; M], offset
 /// * `dest` - Destination array
 /// * `src` - Source array
 /// * `offset` - Offset in source array 
-pub(crate) fn memcpy<N, M>(dest: &mut [u8; N], src: [u8; M], offset: u64) {
-    memcpy_up_to_length(dest, src, offset, N);
+pub(crate) fn memcpy<TItem, SRC_LEN, DEST_LEN>(dest: &mut [TItem; DEST_LEN], src: [TItem; SRC_LEN], offset: u64) {
+    memcpy_up_to_length(dest, src, offset, DEST_LEN);
 }
 
-pub fn sub_array_equals_up_to_length<T, N, M>(
-    subarray: [T; N],
-    array: [T; M],
+pub fn subarray<TItem, SRC_LEN, DEST_LEN>(src: [TItem; SRC_LEN], offset: u64) -> [TItem; DEST_LEN] {
+    let mut dest = [zeroed(); DEST_LEN];
+    memcpy(&mut dest, src, offset);
+    dest
+}
+
+pub fn sub_array_equals_up_to_length<TItem, SUBARRAY_LEN, ARRAY_LEN>(
```

Comment:
```suggestion
pub fn subarray_equals_up_to_length<TItem, SUBARRAY_LEN, ARRAY_LEN>(
```

Reply:
Agree, but I propose to do this refactor in a separate PR because it will increase the size of this PR several times